### ISSUE TITLE
`--isolatedModules` should warn about re-export of type

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#25705]
This was reported originally in https://github.com/Microsoft/TypeScript/issues/15231, but seems to be broken currently.

**Code**

**a.ts**

```ts
export const A = 0;
export type B = number;
```

**b.ts**

```ts
export { A, B } from "./a";
```

**Expected behavior:**

An error saying that the reexports of types are not allowed with `--isolatedModules`

**Actual behavior:**

No error

**Comment by andy-ms** [comment]
@mhegazy I'm seeing the error diagnostic with `typescript@next`.
```sh
> ls
a.ts  b.ts
> cat a.ts
export const A = 0;
export type B = number;
> cat b.ts
export { A, B } from "./a";
> tsc --version
Version 3.0.0-dev.20180712
> tsc a.ts b.ts --isolatedModules
b.ts:1:13 - error TS1205: Cannot re-export a type when the '--isolatedModules' flag is provided.

1 export { A, B } from "./a";
              ~
```

**Comment by mhegazy** [comment]
I swear it was not showing for me yesterday.. maybe i got my self in a bad state with the project.
